server/src/main/java/io/crate/execution/engine/collect/CollectTask.java

```diff
@@ -143,13 +144,14 @@ private void releaseResources() {
     }
 
     @Override
-    public void kill(Throwable throwable) {
+    public void kill(@Nonnull Throwable throwable) {
         if (started.compareAndSet(false, true)) {
             consumer.accept(null, throwable);
         } else {
             batchIterator.whenComplete((it, err) -> {
                 if (err == null) {
                     it.kill(throwable);
+                    consumer.accept(null, throwable);
```

Comment:
This would break the contract that a consumer is only triggered once. 

Depending on the consumer implementation this could then result in duplicate error handling getting executed.